crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -379,30 +379,30 @@ async fn price_improvement_fee_sell_order_test(web3: Web3) {
     // 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
     //
     // Quote: 10000000000000000000 GNO for 9871580343970612988 DAI with
-    // 294580438010728 GNO fee. Equivalent to: (10000000000000000000 +
-    // 294580438010728) GNO for 9871580343970612988 DAI, then scaled to sell amount
-    // gives 10000000000000000000 GNO for 9871289555090525964 DAI
+    // 294580438010728 GNO fee. Equivalent to: 10000000000000000000 GNO for
+    // 9871580343970612988 * ((10000000000000000000 - 294580438010728) /
```

Comment:
Note how we no longer add fee on top of sell amount, but instead, we now reduce the buy amount by converted amount.